nmt.go

```diff
@@ -96,6 +99,7 @@ type NamespacedMerkleTree struct {
 // and for the given namespace size (number of bytes).
 // If the namespace size is 0 this corresponds to a regular non-namespaced
 // Merkle tree.
+// TODO [Me] Why not using a fixed namespace size? i.e., the DefaultNamespaceIDLen, is the variable size for the space efficiency?
```

Comment:
We might move to a larger size in the future. Also, the idea was that this library could also be used independent from Celestia's specific use-case and that an NMT could be useful by itself. That said, I am not of aware of any use case outside of celestia currently.